coderd/notifications/notifier.go

```diff
@@ -226,13 +240,44 @@ func (n *notifier) deliver(ctx context.Context, msg database.AcquireNotification
 			return ctx.Err()
 		default:
 			logger.Debug(ctx, "message dispatch succeeded")
-			success <- newSuccessfulDispatch(n.id, msg.ID)
+			success <- n.newSuccessfulDispatch(msg)
 		}
 	}
+	n.metrics.PendingUpdates.Set(float64(len(success) + len(failure)))
```

Comment:
I think it would be cleaner to just compute this metric on-`Collect`, rather than `Set`ing it when something reads or writes to these channels.  As an illustration of how easy it is to miss a spot, note that you have forgotten to set it in `process()` after a failed call to `prepare()`.